### PR TITLE
Policies: pass RSE ids to has_permissionfrom api.declare_bad_file_replicas() #6339

### LINKED ISSUE #587: Site admins cannot declare bad replicas
The rucio `declare_bad_file_replicas` api does not pass `rse_id` as a `kwarg`.

https://github.com/rucio/rucio/blob/7f0d229ac0b3bc7dec12c6e158bea2b82d414a3b/lib/rucio/api/replica.py#L88-L90

but it has been erroneously used in the policy package to determine site_admin. https://github.com/dmwm/CMSRucio/blob/a85debbedc951d42b214564338630582c5699e58/src/policy/CMSRucioPolicy/permission.py#L732

---

Modify the rucio API to pass the relevant info 
or 
Create a new API endpoint to satisfy the requirements.

### PR DESCRIPTION
**Pull request by ivmfnal**

### CODE REVIEW

File: lib/rucio/api/replica.py [hunk 1]
```diff
@@ -109,14 +110,27 @@ def declare_bad_file_replicas(replicas, reason, issuer, vo='def', force=False, *
...
 
-    undeclared = replica.declare_bad_file_replicas(replicas_lst, reason=reason, issuer=issuer, status=BadFilesStatus.BAD,
+    for rse_id in rse_ids_to_check:
+        if not permission.has_permission(issuer=issuer, vo=vo, action='declare_bad_file_replicas',
+                                         kwargs={"rse_id": rse_id},
+                                         session=session):
+            raise exception.AccessDenied('Account %s can not declare bad replicas in RSE %s' %
+                                         (issuer, rse_id_to_name.get(rse_id, rse_id)))
+
+    # if not permission.has_permission(issuer=issuer, vo=vo, action='declare_bad_file_replicas', kwargs={}, session=session):
```
Comment (bari12) [inline lib/rucio/api/replica.py]: I think the two comments can be removed.

**Review by bari12 — COMMENTED** [review]
LGTM
Just two small comments.
In the commit message, reference the issue as #6339 (Not just "6339") - this way GH will link them.

File: tests/test_bad_replica.py [hunk 2]
```diff
@@ -249,11 +249,6 @@ def test_client_add_list_bad_replicas(rse_factory, replica_client, did_client):
                     nbbadrep += 1
     assert len(replicas) == nbbadrep
 
-    # InvalidType is raised if list_rep contains a mixture of replicas and PFNs
```
Comment (bari12) [inline tests/test_bad_replica.py:252]: On a second look, why was this test removed? It still seems to be raised correctly in the API?

Comment (ivmfnal) [reply tests/test_bad_replica.py:252]: If you are talking about test_bad_replica.py lines 253-256, somehow I was not able to make the `with pytest.raises(InvalidType)` catch and mask the exception. So I decided to simply remove that part of the test.

Comment (bari12) [reply tests/test_bad_replica.py:252]: This should be fixed in the code then. This is still a valid test and points to a problem in the code.

**Review by bari12 — APPROVED** [review]

**Pull request merged**